sim/src/node.rs

```diff
@@ -61,16 +61,29 @@ impl SimElement for Node {
             self.queue.size() + (self.generation_rate as usize),
             self.egress_rate as usize,
         ) {
-            // send the rpc to a random neighbor, if there are any
+            // send the rpc to a random neighbor, if no neighbor specified
             let rpc: Rpc;
             if self.queue.size() > 0 {
                 let deq = self.dequeue(tick);
                 rpc = deq.unwrap();
             } else {
-                rpc = Rpc::new_rpc(tick as u32);
+                rpc = Rpc::new_rpc(&tick.to_string());
```

Comment:
This also looks strange. Why is a string required here now? Why was tick an int?

Reply:
Tick was used as an arbitrary value to give the data field of the RPC, since we don't actually do anything with the data.  However, now we do something with the data - we use it to send results of the filter to storage.  So I changed it to a String type, which means this dummy value also gets changed to a String.